src/schedule/task.c

```diff
@@ -76,6 +76,8 @@ void task_main_init(void)
 
 	*main_task = rzalloc(SOF_MEM_FLAG_KERNEL,
 			     sizeof(**main_task));
+	if (!*main_task)
+		panic(SOF_IPC_PANIC_MEM);
```

Comment:
oh, interesting. I don't see `panic()` defined anywhere! You probably meant `sof_panic()`. We have one more use of `panic()` in https://github.com/thesofproject/sof/blob/8e4079547008c76878887344eccb83b502e16f2a/posix/include/rtos/spinlock.h#L94 - probably macro never used